### ISSUE TITLE
How to use typescript type definitions with umd module

### ISSUE THREAD

**Issue by valeriob** [jakearchibald/idb#94]
Hi,
i'm trying to use idb in a javascript project without modules, so i'm using the build/idb.js artifact.
Inside my .ts file i'm trying to use the library but the typescript project system does not find the types.

I've tryied imorting the definition by this statement in my .ts file
/// <reference path="../../../node_modules/idb/lib/idb.d.ts" />
and by adding the name of the library in the compilerOptions section of the tsconfig.json file.
"types": [
      "idb"
    ],

I'm only able to use the type definitions by using the import syntaxt import { openDb } from "../../../node_modules/idb/lib/idb"; but it can't be used with the compile option "module":"none";

Thanks

**Comment by jakearchibald** [comment]
Thanks for reporting. I'll think about fixes for this, but you can work around it by doing this:

```ts
interface IDB {
  openDB: typeof import('idb').openDB;
  deleteDB: typeof import('idb').deleteDB;
  wrap: typeof import('idb').wrap;
  unwrap: typeof import('idb').unwrap;
}

const idb = self.idb as IDB;
```

**Comment by jakearchibald** [comment]
What build system do you use? Does it support commonjs?

**Comment by valeriob** [comment]
I'm just using the typescript compiler with those options : 
{
  "compileOnSave": true,
  "compilerOptions": {
    "noEmitOnError": true,
    "noImplicitAny": false,
    "removeComments": false,
    "sourceMap": true,
    "target": "es6",
    "module": "none",
    "skipLibCheck": true
  },
  "exclude": [
    "bin",
    "obj"
  ]
}

I was able to make it work with those import declarations : 

import { DB } from '../../../node_modules/idb/lib/idb';
declare var idb: typeof import('../../../node_modules/idb/lib/idb');

Basically the first one just lights up types if used directly (like if you want to do let db: DB), the second line allows to write this code :
let db = async idb.openDb("dbName");

The triky part is that when you import more than interfaces from this statement 
import { DB, openDb } from '../../../node_modules/idb/lib/idb';
the typescript compiler switches to modules mode and gives an error that modules are disabled (i want the out of my way).

**Comment by jakearchibald** [comment]
Out of interest, what's stopping you using modules with TypeScript?

**Comment by valeriob** [comment]
We build LOB web application, and use many mini vuejs application in the same project (most pages are server rendered, and some are client side application), and we do not want to deal with the complexity with bundlers, when a simple page include is sufficient.

**Comment by valeriob** [comment]
In addition module systems are a real mess and a pain to deal with for many other reasons:
- unstable
- complicated
- slow to compile
- need heavy/slow tooling

We just save a file and reload the page, no tooling needed, just typescript compiler.

**Comment by jakearchibald** [comment]
Thanks for the info!
